src/python/pants/backend/explorer/BUILD

```diff
@@ -21,3 +21,15 @@ python_tests(
     name="tests",
     dependencies=[":explorer"],
 )
+
+
+__dependents_rules__(
+    # Block any outside sources from pulling in code from this backend, as it is self contained and
+    # have 3rd party dependencies that we don't want to leak to the core Pants distribution.
+    (
+        "*",
+        build_file_dir(),
```

Comment:
Not directly related to this change, but this is a bit ugly. "Allow things in and under this dir to depend on other things in this dir" seems like a common case, maybe we should have syntactic sugar for it? E.g., maybe `.` means "this BUILD file's dir" and `::` means "this BUILD file's dir and all its subdirs", so

`('*', '::', '!*')`

Reply:
Ah... nice, I like the re-use of the `::` pattern.

Reply:
although, from the address spec globs `::` means all targets, where as here it would need to be `./::` to match the semantics of "this dir and below".
But then there's the `.` that would move to where it is applied.. looking at targets we have a single `:` for naming targets in the current dir, so what if `:/some/path` would mean from this build file.. or `:/*` for all files in this build files sub tree.
So the common pattern for "this build files dir and all sub directories" would be `":", ":/*"`.

Here, I think using `::` may come in handy as a replacement for `*` to mean the named directory and all sub dirs, so `some/path/::` is the same as `some/path::` which translates to `"some/path", "some/path/*"`.

Reply:
Yeah, probably what we actually want is `**/*` to mean "this dir and all subdirs", see my comment on the discussion

Reply:
So `**/*` as a fixed thing? or could you say just `**` and what would that imply?

Btw, opened draft PR for the `:` and `::` alternative: #17588

Reply:
Let's discuss there then